micro_sam/models/peft_sam.py

```diff
@@ -141,6 +142,54 @@ def forward(self, x):
         return x
 
 
+class AdaptFormer(nn.Module):
+    """Adds AdaptFormer Module in place of the MLP Layers
+
+    Args:
+        rank: The rank here refers to the size of the up and down projection layers
+        block: The chosen encoder block for implementing AdaptFormer
+        alpha: A parameters that scales the Adapter path. Can be either learnable or some fixed value.
+    """
+    def __init__(self, rank: int, block: nn.Module, alpha: str = '0.1'):
+        super().__init__()
+
+        self.mlp_proj = block.mlp
+        self.n_embd = block.mlp.lin1.in_features
+
+        if alpha == 'learnable_scalar':
+            self.alpha = nn.Parameter(torch.ones(1))
+        else:
+            self.alpha = float(alpha)
+
+        self.rank = rank
```

Comment:
@anwai98 This is technically not a rank as in LoRA but it determines the size of the adapter layers. The authors of AdaptFormer refer to this as 'bottleneck middle dimension'. Since the impact of the parameter is similiar to a LoRA rank, I propose naming it rank to keep the number of different peft hyperparameter names as low as possible.

Reply:
I would keep the connotation of `rank` only limited to matrix decomposition and call this new argument as: `projection_size` or something like that (+ this is also why we have `module_kwargs` in `PEFT_Sam`, which capture all additional variables).

PS. This is however details which we can refactor later. It's important to validate these methods first.

Reply:
Ok I'll take care of it